Tetrahedron_3_Triangle_3 intersection on clang

There's a difference between GCC and Clang, https://clang.llvm.org/compatibility.html#dep_lookup,  revealed by an issue in CGAL. 

## Issue Details

```
In file included from /Users/goretkin/projects/CGAL.jl/deps/cgal/Intersections_3/include/CGAL/Intersections_3/Triangle_3_Triangle_3.h:19:
In file included from /Users/goretkin/projects/CGAL.jl/deps/cgal/Intersections_3/include/CGAL/Intersections_3/internal/Triangle_3_Triangle_3_intersection.h:18:
In file included from /Users/goretkin/projects/CGAL.jl/deps/cgal/Intersections_3/include/CGAL/Intersections_3/Line_3_Triangle_3.h:20:
In file included from /Users/goretkin/projects/CGAL.jl/deps/cgal/Intersections_3/include/CGAL/Intersections_3/internal/Triangle_3_Line_3_intersection.h:19:
In file included from /Users/goretkin/projects/CGAL.jl/deps/cgal/Intersections_3/include/CGAL/intersections.h:38:
In file included from /Users/goretkin/projects/CGAL.jl/deps/cgal/Intersections_3/include/CGAL/intersection_3.h:31:
In file included from /Users/goretkin/projects/CGAL.jl/deps/cgal/Intersections_3/include/CGAL/Intersections_3/Bbox_3_Tetrahedron_3.h:20:
/Users/goretkin/projects/CGAL.jl/deps/cgal/Intersections_3/include/CGAL/Intersections_3/internal/Tetrahedron_3_Bounded_3_do_intersect.h:57:25: error: call to function 'do_intersect' that is neither visible in the template definition nor found by argument-dependent lookup
      const Boolean b = do_intersect(tr,
                        ^
/Users/goretkin/projects/CGAL.jl/deps/cgal/Intersections_3/include/CGAL/Intersections_3/internal/Tetrahedron_3_Bounded_3_do_intersect.h:79:10: note: in instantiation of function template specialization 'CGAL::Intersections::internal::do_intersect_tetrahedron_bounded<CGAL::Simple_cartesian<CGAL::Quotient<CGAL::MP_Float> >, CGAL::Triangle_3<CGAL::Simple_cartesian<CGAL::Quotient<CGAL::MP_Float> > > >' requested here
  return do_intersect_tetrahedron_bounded(tr, tet, tr[0], k);
         ^
/Users/goretkin/projects/CGAL.jl/deps/cgal/Kernel_23/include/CGAL/Kernel/function_objects.h:3009:39: note: in instantiation of function template specialization 'CGAL::Intersections::internal::do_intersect<CGAL::Simple_cartesian<CGAL::Quotient<CGAL::MP_Float> > >' requested here
    { return Intersections::internal::do_intersect(t1, t2, K()); }
                                      ^
/Users/goretkin/projects/CGAL.jl/deps/cgal/Filtered_kernel/include/CGAL/Filtered_predicate.h:106:12: note: in instantiation of function template specialization 'CGAL::CommonKernelFunctors::Do_intersect_3<CGAL::Simple_cartesian<CGAL::Quotient<CGAL::MP_Float> > >::operator()<CGAL::Tetrahedron_3<CGAL::Simple_cartesian<CGAL::Quotient<CGAL::MP_Float> > >, CGAL::Triangle_3<CGAL::Simple_cartesian<CGAL::Quotient<CGAL::MP_Float> > > >' requested here
    return ep(c2e(args)...);
           ^
/Users/goretkin/projects/CGAL.jl/deps/cgal/Intersections_3/include/CGAL/Intersections_3/Tetrahedron_3_Triangle_3.h:23:3: note: in instantiation of function template specialization 'CGAL::Filtered_predicate<CGAL::CommonKernelFunctors::Do_intersect_3<CGAL::Simple_cartesian<CGAL::Quotient<CGAL::MP_Float> > >, CGAL::CommonKernelFunctors::Do_intersect_3<CGAL::Simple_cartesian<CGAL::Interval_nt<false> > >, CGAL::Cartesian_converter<CGAL::Type_equality_wrapper<CGAL::Cartesian_base_no_ref_count<double, CGAL::Epick>, CGAL::Epick>, CGAL::Simple_cartesian<CGAL::Quotient<CGAL::MP_Float> >, CGAL::NT_converter<double, CGAL::Quotient<CGAL::MP_Float> > >, CGAL::Cartesian_converter<CGAL::Type_equality_wrapper<CGAL::Cartesian_base_no_ref_count<double, CGAL::Epick>, CGAL::Epick>, CGAL::Simple_cartesian<CGAL::Interval_nt<false> >, CGAL::NT_converter<double, CGAL::Interval_nt<false> > >, true>::operator()<CGAL::Tetrahedron_3<CGAL::Epick>, CGAL::Triangle_3<CGAL::Epick> >' requested here
  CGAL_DO_INTERSECT_FUNCTION(Tetrahedron_3, Triangle_3, 3)
  ^
/Users/goretkin/projects/CGAL.jl/deps/cgal/Intersections_2/include/CGAL/Intersection_traits.h:67:25: note: expanded from macro 'CGAL_DO_INTERSECT_FUNCTION'
    return BOOST_PP_CAT(K().do_intersect_, BOOST_PP_CAT(DIM, _object()(a, b))); \
                        ^
/Cxx.h:1:1: note: in instantiation of function template specialization 'CGAL::do_intersect<CGAL::Epick>' requested here
/Users/goretkin/projects/CGAL.jl/deps/cgal/Intersections_3/include/CGAL/Intersections_3/internal/Triangle_3_Triangle_3_do_intersect.h:198:1: note: 'do_intersect' should be declared prior to the call site or in an associated namespace of one of its arguments
do_intersect(const typename K::Triangle_3 &t1,
```

## Source Code

```

#include <CGAL/Exact_predicates_inexact_constructions_kernel.h>

typedef CGAL::Exact_predicates_inexact_constructions_kernel K;
typedef K::Point_3          Point_3;
typedef K::Vector_3         Vector_3;
typedef K::Triangle_3       Triangle_3;
typedef K::Tetrahedron_3    Tetrahedron_3;

int main( int argc, const char* argv[] )
{
  Tetrahedron_3 tet1(Point_3(1,-1,0), Point_3(-1,-1,0), Point_3(0,1,0), Point_3(0,0,1));
  Triangle_3 tri4(Point_3(1,-1,0.5), Point_3(-1,-1,0.5), Point_3(0,1,0.5));

  std::cout << CGAL::do_intersect(tet1, tri4) << std::endl;
}
```

## Environment

* Operating system (Windows/Mac/Linux, 32/64 bits): macOS
* Compiler: clang
* CGAL version: master

--- a/Intersections_3/include/CGAL/Intersections_3/internal/Triangle_3_Line_3_intersection.h
+++ b/Intersections_3/include/CGAL/Intersections_3/internal/Triangle_3_Line_3_intersection.h
@@ -24,8 +24,7 @@
 #ifndef CGAL_INTERNAL_INTERSECTIONS_3_TRIANGLE_3_LINE_3_INTERSECTION_H
 #define CGAL_INTERNAL_INTERSECTIONS_3_TRIANGLE_3_LINE_3_INTERSECTION_H
 
-#include <CGAL/kernel_basic.h>
-#include <CGAL/intersections.h>
+#include <CGAL/Intersections_3/Line_3_Plane_3.h>
 
 namespace CGAL {
   

